spring-cloud-dataflow-aggregate-task/src/main/java/org/springframework/cloud/dataflow/aggregate/task/impl/AggregateDataFlowTaskExecutionQueryDao.java

```diff
@@ -310,6 +361,27 @@ public long getTaskExecutionCountByTaskName(String taskName) {
 		return count != null ? count : 0L;
 	}
 
+	@Override
+	public long getTaskExecutionCountByTaskNameAndBeforeDate(String taskName, Date endTime) {
+		Long count;
+		if (StringUtils.hasText(taskName)) {
+			final SqlParameterSource queryParameters = new MapSqlParameterSource()
+					.addValue("taskName", taskName, Types.VARCHAR)
+					.addValue("endTime", endTime, Types.DATE);
+
+			try {
+				count = this.jdbcTemplate.queryForObject(TASK_EXECUTION_COUNT_BY_NAME_AND_BEFORE_END_TIME, queryParameters, Long.class);
+			} catch (EmptyResultDataAccessException e) {
+				count = 0L;
```

Comment:
I am curious why the later `queryForObject` below is also not in a try/catch to enforce return 0 default? I see there are other cases in this code that are already doing the same.

Reply:
Thanks - fixed. 👍